Connected Profile Page to Firestore

wyatt-js opened the pull request:
# Summary
[Backend: Create User Document Upon Signup](https://github.com/cssgunc/Parr-center-refactor/issues/28) & [BackEnd: Profile Page](https://github.com/cssgunc/Parr-center-refactor/issues/25) are complete
Profile page now shows relevant user data, and a document for a user is created in the users collection upon signup.
Users now have isAdmin booleans tied to them.

# Changes
Added three functions in dp-operations 

1. To check if user does not exist in collections
2. Create user in collections
3. Check admin

Added isAdmin to user type.
Added admin badge to user profile page.

Jordweinstein commented on frontend/src/lib/firebase/types.ts:
```diff
@@ -10,6 +10,7 @@ export interface User {
   photoURL?: string;
   createdAt: Date;
```
For createdAt and lastLogin, allow the User type to accept Firestore Timesstamp || Date since createUser writes serverTimestamp(). You could also convert timestamp to date upon read.

Jordweinstein reviewed (changes requested):
Awesome work!

Some changes: 
1. Fix the createdAt / lastLoginAt type mismatch (I left a comment re. this)
2. Add/verify Firestore security rules so only trusted server processes or existing admins can set/modify isAdmin.
3. Action: update EmailAuth registerUser and any other place using name[0].
4. Use setDoc(userDocRef, {...}, { merge: true }) to avoid overwriting any existing fields unintentionally
5. Make sure you're using correct variable names/casing (displayname or displayName)

wyatt-js commented:
- Made profile picture a next.js Image element to prevent google rate limiting
- Added respective links to next.js config
- Fixed time types
- Fixed update user in UserAuth for profile picture and username
- Added merge: true to setDoc
- Updated naming conventions
- Given admin liberal privileges in firestore rules to CRUD modules
